### PR TITLE
Use dio networking manager for image upload

### PR DESCRIPTION
**Pull request by n13**
Fixes: https://github.com/hypha-dao/hypha_wallet/issues/88

1 - use dio networking manager

2 - clean up the code - _postImage is now inside amplify service, and the policy object in its own file.

Tested - it works. I am able to upload new images, and delete images on the profile.

### CODE REVIEW

File: lib/core/network/api/aws_amplify/amplify_policy.dart
Comment (gguijarro-c-chwy) [inline lib/core/network/api/aws_amplify/amplify_policy.dart:1]: Is this a model? Should we put this file with other models?

File: lib/core/network/api/aws_amplify/amplify_policy.dart [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    String bucket,
+    int expiryMinutes,
+    String accessKeyId,
+    int maxFileSize,
+    String sessionToken, {
+    required String region,
+  }) {
+    final datetime = SigV4.generateDatetime();
+    final expiration = DateTime.now().add(Duration(minutes: expiryMinutes)).toUtc().toString().split(' ').join('T');
+    // print('expiration: $expiration');
```
Comment (gguijarro-c-chwy) [inline lib/core/network/api/aws_amplify/amplify_policy.dart:38]: needed?

**Review by gguijarro-c-chwy — APPROVED** [review]
PR looks good. some very minor things

File: lib/core/network/api/aws_amplify/amplify_policy.dart
Comment (n13) [reply lib/core/network/api/aws_amplify/amplify_policy.dart:1]: No, I want to keep aws stuff contained here inside its folder, all these classes belong together and all they do is allow us to make aws calls.

File: lib/core/network/api/aws_amplify/amplify_policy.dart [hunk 2]
```diff
@@ -0,0 +1,67 @@
+// Note: This code based on the ewxample in amazon_cognito_identity_dart_2 flutter package
```
[the remark this answers is not in the document]
Comment (n13) [reply lib/core/network/api/aws_amplify/amplify_policy.dart]: Fixed

**Comment by n13** [comment]
More cleanup, thanks Gery.

**Pull request merged**